examples/keras_spark3_rossmann.py

```diff
@@ -554,11 +587,18 @@ def fn(rows):
         import tensorflow as tf
         import tensorflow.keras.backend as K
 
-        # Do not use GPUs for prediction, use single CPU core per task.
-        config = tf.ConfigProto(device_count={'GPU': 0})
-        config.inter_op_parallelism_threads = 1
-        config.intra_op_parallelism_threads = 1
-        K.set_session(tf.Session(config=config))
+        if GPU_INFERENCE_ENABLED:
+            from pyspark import TaskContext
+            config = tf.ConfigProto()
+            config.gpu_options.allow_growth = True
+            config.gpu_options.visible_device_list = TaskContext.get().resources()['gpu'].addresses[0]
```

Comment:
You should use: get_available_devices()[0]
See examples/keras_spark3_rossmann.py for a reference

Reply:
Thanks for reviewing. But I don't think that will work.

For training, "horovod.spark.run(train_fn, args, ...)" is invoked, which will setup the whole environment (e.g., set GPU resources for each worker) and trigger distributed training. That's why you could call "horovod.spark.task.get_available_devices" in each worker.

For transform, it's just a pure Spark application. You cannot call "horovod.spark.task.get_available_devices", because Horovod initialization is not involved and "horovod.spark.task.task_info.set_resources" is not even invoked. Now that it's a pure Spark application, why not get GPU resources in Spark way by calling "TaskContext.get().resources()['gpu']"?

@tgaddair, please correct me if I were wrong, thanks in advance.

Reply:
Yes, you're correct @chuanlihao.  For inference we are not running Horovod, just using a Spark UDF.  I would add a comment explaining this distinction, because it's not obvious.

Reply:
Thanks for your confirmation. Added a comment for this distinction in line 601, commit 4.